### PR TITLE
Openscad and its dependencies

### PR DESCRIPTION
**Pull request by lyzech**
openscad is is a software for creating solid 3D CAD objects (http://www.openscad.org).

These templates are for Qt5 only. qscintilla and openscad could be easily packaged with Qt4.

### CODE REVIEW

File: srcpkgs/opencsg/template [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+license="GPL-2"
+homepage="http://www.opencsg.org"
+distfiles="http://www.opencsg.org/OpenCSG-${version}.tar.gz"
+checksum="d952ec5d3a2e46a30019c210963fcddff66813efc9c29603b72f9553adff4afb"
+wrksrc="OpenCSG-${version}"
+
+makedepends="glew-devel libfreeglut-devel"
+
+do_build() {
+	cd src
```
Comment (Vaelatern) [inline srcpkgs/opencsg/template]: `build_wrksrc` maybe?

File: srcpkgs/openscad/template [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+license="GPL-2"
+homepage="http://www.openscad.org"
+distfiles="http://files.openscad.org/${pkgname}-${_distver}.src.tar.gz"
+checksum="a2535bb9e27d96e10e1e19268a596155164c7129d410a0c7f96edc2d09400083"
+wrksrc="${pkgname}-${_distver}"
+
+hostmakedepends="bison flex pkg-config qt5-qmake"
+makedepends="cgal-devel harfbuzz-devel opencsg-devel qscintilla-qt5-devel"
+#hostmakedepends="flex bison pkg-config qt5-qmake"
+#makedepends="boost-devel cgal-devel eigen fontconfig-devel freetype-devel glew-devel gmp-devel mpfr-devel harfbuzz-devel opencsg-devel qscintilla-qt5-devel qt5-devel"
```
Comment (Vaelatern) [inline srcpkgs/openscad/template]: Comments not necessary?

File: srcpkgs/qscintilla-qt5/template [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+do_install() {
+	cd Qt4Qt5
+	make STRIP=true PREFIX=/usr DESTDIR=${DESTDIR} \
+		INSTALL_ROOT=${DESTDIR} install
+	# designer plugin
+	#cd ../designer-Qt4Qt5
+	#make INSTALL_ROOT=${DESTDIR} install
+}
```
Comment (Vaelatern) [inline srcpkgs/qscintilla-qt5/template]: Any reason you couldn't use an appropriate `build_style` with these, and `build_wrksrc`?

File: srcpkgs/opencsg/template (hunk 1 above)
Comment (lyzech) [inline srcpkgs/opencsg/template]: Yes, thank you.

File: srcpkgs/qscintilla-qt5/template (hunk 3 above)
Comment (lyzech) [inline srcpkgs/qscintilla-qt5/template]: I originally wanted to build all the plugins and bindings, but they are useless. So I corrected the template according your suggestions.

**Comment by lyzech** [comment]
Thank you for your review. I amended my commits using your suggestions.

**Comment by karl-nilsson** [comment]
@Vaelatern I successfully built the openscad package and dependencies (x64, gnu libc), and ran through all the examples without a problem. lgtm

**Pull request closed without merging**